data/state/accountsDB_test.go

```diff
@@ -1270,3 +1270,72 @@ func TestAccountsDB_MainTrieAutomaticallyMarksCodeUpdatesForEviction(t *testing.
 	hashesForEviction = ewl.Cache[string(rootHash2)]
 	assert.Equal(t, 3, len(hashesForEviction))
 }
+
+func TestAccountsDB_RemoveAccountSetsObsoleteHashes(t *testing.T) {
+	t.Parallel()
+
+	marshalizer := &mock.MarshalizerMock{}
+	hsh := mock.HasherMock{}
+	db := mock.NewMemDbMock()
+	ewl, _ := mock.NewEvictionWaitingList(100, mock.NewMemDbMock(), marshalizer)
+	storageManager, _ := trie.NewTrieStorageManager(db, marshalizer, hsh, config.DBConfig{}, ewl, config.TrieStorageManagerConfig{})
+	maxTrieLevelInMemory := uint(5)
+	tr, _ := trie.NewTrie(storageManager, marshalizer, hsh, maxTrieLevelInMemory)
+	adb, _ := state.NewAccountsDB(tr, hsh, marshalizer, factory.NewAccountCreator())
+
+	addr := make([]byte, 32)
+	acc, _ := adb.LoadAccount(addr)
+	userAcc := acc.(state.UserAccountHandler)
+	userAcc.DataTrieTracker().SaveKeyValue([]byte("key"), []byte("value"))
+
+	_ = adb.SaveAccount(userAcc)
+	_, _ = adb.Commit()
+
+	acc, _ = adb.LoadAccount(addr)
+	userAcc = acc.(state.UserAccountHandler)
+	userAcc.SetCode([]byte("code"))
+	snapshot := adb.JournalLen()
+
+	err := adb.RemoveAccount(addr)
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(adb.GetObsoleteHashes()))
+
+	err = adb.RevertToSnapshot(snapshot)
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(adb.GetObsoleteHashes()))
+}
+
+func TestAccountsDB_RemoveAccountMarksObsoleteHashesForEviction(t *testing.T) {
+	t.Parallel()
+
+	marshalizer := &mock.MarshalizerMock{}
+	hsh := mock.HasherMock{}
+	db := mock.NewMemDbMock()
+	ewl, _ := mock.NewEvictionWaitingList(100, mock.NewMemDbMock(), marshalizer)
+	storageManager, _ := trie.NewTrieStorageManager(db, marshalizer, hsh, config.DBConfig{}, ewl, config.TrieStorageManagerConfig{})
+	maxTrieLevelInMemory := uint(5)
+	tr, _ := trie.NewTrie(storageManager, marshalizer, hsh, maxTrieLevelInMemory)
+	adb, _ := state.NewAccountsDB(tr, hsh, marshalizer, factory.NewAccountCreator())
+
+	addr := make([]byte, 32)
+	acc, _ := adb.LoadAccount(addr)
+	userAcc := acc.(state.UserAccountHandler)
+	userAcc.DataTrieTracker().SaveKeyValue([]byte("key"), []byte("value"))
+	_ = adb.SaveAccount(userAcc)
+
+	addr1 := make([]byte, 32)
+	addr1[0] = 1
+	acc, _ = adb.LoadAccount(addr1)
+	_ = adb.SaveAccount(acc)
+
+	rootHash, _ := adb.Commit()
+
+	err := adb.RemoveAccount(addr)
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(adb.GetObsoleteHashes()))
```

Comment:
assert also the equality, not just length 1

Reply:
Done.